geotrek/land/models.py

```diff
@@ -360,3 +360,79 @@ def log_cascade_deletion_from_signagemanagementedge_topology(sender, instance, u
 Project.add_property('signage_edges', lambda self: self.edges_by_attr('signage_edges'), _("Signage management edges"))
 if 'geotrek.signage' in settings.INSTALLED_APPS:
     Blade.add_property('signage_edges', lambda self: self.signage.signage_edges, _("Signage management edges"))
+
+
+class CirculationType(StructureOrNoneRelated):
+    name = models.CharField(max_length=128, verbose_name=_("Name"))
+
+    class Meta:
+        verbose_name = _("Circulation type")
+        verbose_name_plural = _("Circulation types")
+        ordering = ['name']
+
+    def __str__(self):
+        if self.structure:
+            return "{} ({})".format(self.name, self.structure.name)
+        return self.name
+
+
+class CirculationEdge(GeotrekMapEntityMixin, Topology):
+    topo_object = models.OneToOneField(Topology, parent_link=True, on_delete=models.CASCADE)
+    circulation_type = models.ForeignKey(CirculationType, verbose_name=_("Circulation type"), on_delete=models.PROTECT)
+    authorized = models.BooleanField(verbose_name=_("Authorized"), default=None, null=True)
+    eid = models.CharField(verbose_name=_("External id"), max_length=1024, blank=True, null=True)
```

Comment:
```suggestion
    eid = models.CharField(verbose_name=_("External id"), max_length=1024, blank=True)
```
It is recommended not to have both `null=True` and `blank=True` or `CharFields` : https://steelkiwi.com/blog/best-practices-working-django-models-python/ (number 22)
Here we need to allow `null` as value for undefined 

However, I'm still not able to set circulation to "undefined" even after this change and recreating the migration, the form is accepted but doesn't change the value, I think we need to find a way to make `NullBooleanSelect` work, or something equivalent

Reply:
I removed `null=True`, which was directly copied from another model already existing. I checked and roughly 85 fields in various geotrek models also have `blank=True, null=True`, maybe that's a thing to verify sometime